File: csharp/Lift/LiftSystem.cs
using System.Collections.Generic;
using System.Linq;

namespace Lift
{
    public class LiftSystem
    {
        public List<int> Floors { get; }
        public List<Lift> Lifts { get; }
        public List<Call> Calls { get; }

        public LiftSystem(List<int> floors, List<Lift> lifts, List<Call> calls)
        {
            Floors = floors;
            Lifts = lifts;
            Calls = calls;
        }

        public List<int> getFloorsInDescendingOrder()
        {
            var copy = new List<int>(this.Floors);
            copy.Reverse();
            return copy;
        }

        public List<Call> getCallsForFloor(int floor)
        {
            return Calls.Where(c => c.Floor == floor).ToList();
        }

        
    }
}

File: php/src/LiftSystem.php
<?php

declare(strict_types=1);

namespace Lift;

class LiftSystem
{
    /**
     * @param int[] $floors
     * @param Call[] $calls
     * @param Lift[] $lifts
     */
    public function __construct(
        private array $floors,
        private array $lifts,
        private array $calls
    ) {
    }

    public function getFloorsInDescendingOrder(): array
    {
        return array_reverse($this->floors);
    }

    public function getCallsForFloor(int $floor): array
    {
        return array_filter($this->calls, fn ($call) => /** @var Call $call */ $call->getFloor() === $floor);
    }

    public function getLifts(): array
    {
        return $this->lifts;
    }

    public function tick(): void
    {
        // TODO: implement this method
    }
}
